### PR TITLE
ocamlPackages.linol: 0.5 → 0.6

### PR DESCRIPTION
**Pull request by vbgl**
## Description of changes

https://raw.githubusercontent.com/c-cube/linol/main/CHANGES.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/ocaml-modules/linol/default.nix [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
-    owner = "c-cube";
-    repo = "linol";
-    rev = "v${version}";
-    hash = "sha256-ULPOB/hb+2VXDB/eK66WDDh/wj0bOwUt0tZsiIXqndo=";
-  };
-
-  patches = fetchpatch {
-    url = "https://github.com/c-cube/linol/commit/d8ebcf9a60f1e7251d14cdcd0b2ebd5b7f8eec6d.patch";
-    hash = "sha256-JHR0P0X3ep5HvDWW43dMb452/WsFKS4e+5Qhk4MzaxQ=";
+  src = fetchurl {
```
Comment (gador) [inline pkgs/development/ocaml-modules/linol/default.nix:10]: Is there a reason you switched from Github source to the download artefact? (If so, a comment or a note in the commit message would be appreciated)

File: pkgs/development/compilers/ligo/default.nix [hunk 2]
```diff
@@ -24,6 +24,8 @@ ocamlPackages.buildDunePackage rec {
     fetchSubmodules = true;
   };
 
+  patches = [ ./linol.patch ];
```
Comment (gador) [inline pkgs/development/compilers/ligo/default.nix:27]: Please add a comment or name the patch appropiately to indicate why the patch is needed, or what it does.

**Review by gador — CHANGES_REQUESTED** [review]

File: pkgs/development/compilers/ligo/default.nix (hunk 2 above)
Comment (vbgl) [reply pkgs/development/compilers/ligo/default.nix]: Thanks for your careful review. I think your comments have been addressed.

**Review by gador — APPROVED** [review]

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 331337` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>ligo</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
ligo:
<details>

warning: EvalError
Cannot evaluate attribute ‘ligo’ in ‘/nix/store/8yjpp20ymwzhdhc545vyxb5q4vdrix0h-nixpkgs’.
</details>
</details>

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 331337` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>ligo</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
ligo:
<details>

warning: EvalError
Cannot evaluate attribute ‘ligo’ in ‘/nix/store/x25s6g7qxarjsi4jsqcfk3yn3drps6sj-nixpkgs’.
</details>
</details>

**Pull request merged**